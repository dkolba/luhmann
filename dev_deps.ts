export { assertSnapshot } from "https://deno.land/std@0.188.0/testing/snapshot.ts";
