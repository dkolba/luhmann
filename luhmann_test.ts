import { assertSnapshot } from "./dev_deps.ts";
import {
  generateDocument,
  markupify,
  simpleSnippet,
  simpleTemplate,
  teaserifyDocs,
  validateDocs,
} from "./mod.ts";

const frontmatterMarkdownDocument = `---
title: A first test title
author: Boaty McBoatface
date: 2032-05-21
teaser: Best test teaser in the industry
---

## Test Headline 1

An unordered test list

- Test list item 1
- Test list item 2
- Test list item 3

An ordered test list

1. Test list item 4
2. Test list item 5
3. Test list item 6

### A test subheadling

Some test text`;

Deno.test("simpleTemplate isSnapshotMatch", async function (t): Promise<void> {
  const body = simpleTemplate("<span>TEST SPAN TAG</span>", [], "");
  await assertSnapshot(t, body);
});

Deno.test("validateDocs isSnapshotMatch", async function (t): Promise<void> {
  const docResponses = [
    new Response(frontmatterMarkdownDocument),
    new Response("## asdf"),
  ];
  const docList = ["test.md", "asdf.md"];
  const validDocs = await validateDocs(docResponses, docList);
  await assertSnapshot(t, validDocs);
});

Deno.test(
  "simpleSnippet with bare data isSnapshotMatch",
  async function (t): Promise<void> {
    const body = simpleSnippet({
      date: new Date(
        "Mon May 22 2023 21:57:18 GMT+0200 (Central European Summer Time"
      ),
    });
    await assertSnapshot(t, body);
  }
);

Deno.test(
  "simpleSnippet with full data isSnapshotMatch",
  async function (t): Promise<void> {
    const body = simpleSnippet({
      mdName: "testslug",
      title: "Test title",
      teaser: "Test teaser",
      date: new Date(
        "Mon May 22 2023 21:57:18 GMT+0200 (Central European Summer Time"
      ),
    });
    await assertSnapshot(t, body);
  }
);

Deno.test(
  "generateDocument isSnapshotMatch",
  async function (t): Promise<void> {
    const document = generateDocument({
      mdName: "testdoc.md",
      mdContent: frontmatterMarkdownDocument,
    });
    await assertSnapshot(t, document);
  }
);

Deno.test(
  "teaserifyDocs & markupify isSnapshotMatch",
  async function (t): Promise<void> {
    const validDocs = [
      {
        mdName: "test",
        mdContent: frontmatterMarkdownDocument,
      },
      { mdName: "asdf", mdContent: "## asdf" },
    ];
    const teasers = await teaserifyDocs(validDocs);
    await assertSnapshot(t, teasers);

    const markup = teasers.map((teaser) => markupify(teaser, simpleSnippet));
    await assertSnapshot(t, markup);
  }
);
