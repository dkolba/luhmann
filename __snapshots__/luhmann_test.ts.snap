export const snapshot = {};

snapshot[`simpleTemplate isSnapshotMatch 1`] = `
'<!doctype html>
  <html lang=en>
    <head>
      <meta charset=utf-8>
      <title>Zettelkasten</title>
      
      <style>
        main {
          max-width: 800px;
          margin: 0 auto;
        }
        
      </style>
    </head>
    <body>
      <main data-color-mode="light" data-light-theme="light" data-dark-theme="dark" class="markdown-body">
      <span>TEST SPAN TAG</span>
    </main>
  </body>
</html>
'
`;

snapshot[`validateDocs isSnapshotMatch 1`] = `
[
  {
    mdContent: "---
title: A first test title
author: Boaty McBoatface
date: 2032-05-21
teaser: Best test teaser in the industry
---

## Test Headline 1

An unordered test list

- Test list item 1
- Test list item 2
- Test list item 3

An ordered test list

1. Test list item 4
2. Test list item 5
3. Test list item 6

### A test subheadling

Some test text",
    mdName: "test",
  },
  {
    mdContent: "## asdf",
    mdName: "asdf",
  },
]
`;

snapshot[`simpleSnippet with bare data isSnapshotMatch 1`] = `
'<section>
                      <h2><a href="/missing name">missing title</a></h2>
                      <time datetime="Mon May 22 2023 21:57:18 GMT+0200 (Central European Summer Time)"></time>
                      <span>missing teaser</span>
\\t\\t    </section>'
`;

snapshot[`simpleSnippet with full data isSnapshotMatch 1`] = `
'<section>
                      <h2><a href="/testslug">Test title</a></h2>
                      <time datetime="Mon May 22 2023 21:57:18 GMT+0200 (Central European Summer Time)"></time>
                      <span>Test teaser</span>
\\t\\t    </section>'
`;

snapshot[`generateDocument isSnapshotMatch 1`] = `
{
  frontmatter: {
    author: "Boaty McBoatface",
    date: 2032-05-21T00:00:00.000Z,
    teaser: "Best test teaser in the industry",
    title: "A first test title",
  },
  markdown: "## Test Headline 1

An unordered test list

- Test list item 1
- Test list item 2
- Test list item 3

An ordered test list

1. Test list item 4
2. Test list item 5
3. Test list item 6

### A test subheadling

Some test text",
  mdName: "testdoc.md",
}
`;

snapshot[`teaserifyDocs & markupify isSnapshotMatch 1`] = `
[
  {
    frontmatter: {
      author: "Boaty McBoatface",
      date: 2032-05-21T00:00:00.000Z,
      teaser: "Best test teaser in the industry",
      title: "A first test title",
    },
    markdown: "## Test Headline 1

An unordered test list

- Test list item 1
- Test list item 2
- Test list item 3

An ordered test list

1. Test list item 4
2. Test list item 5
3. Test list item 6

### A test subheadling

Some test text",
    mdName: "test",
  },
]
`;

snapshot[`teaserifyDocs & markupify isSnapshotMatch 2`] = `
[
  '<section>
                      <h2><a href="/test">A first test title</a></h2>
                      <time datetime="Fri May 21 2032 02:00:00 GMT+0200 (Central European Summer Time)">21.5.2032</time>
                      <span>Best test teaser in the industry</span>
\\t\\t    </section>',
]
`;
